validators_test.go

```diff
@@ -119,3 +119,42 @@ func TestValidators__validateSimpleTime(t *testing.T) {
 		}
 	}
 }
+
+// Ensure that ASCII characters from 0x20 to 0x7E are considered alphanumeric.
+func TestValidators__isAlphanumeric(t *testing.T) {
+	v := validator{}
+
+	for i := 0; i < 255; i++ {
+		err := v.isAlphanumeric(string(rune(i)))
+
+		if i < 0x20 || i > 0x7E {
+			if err == nil {
+				t.Errorf("expected rune %x to be non-alphanumeric", i)
+			}
+		} else {
+			if err != nil {
+				t.Errorf("expected rune %x to be alphanumeric: %v", i, err)
+			}
+		}
+	}
```

Comment:
I think you can simplify this to:
```go
		err := v.isAlphanumeric(string(rune(i)))
		if (i < 0x20 || i > 0x7E) && err == nil {
			t.Errorf("expected rune %x to be non-alphanumeric", i)
		} else if err != nil {
			t.Errorf("expected rune %x to be alphanumeric: %v", i, err)
		}

```

Reply:
Not quite, because that will braek if i is outside the 0x20 - 0x7E range - in that scenario, we expect an error. But I can still simplify a little.